python/flatbuffers/builder.py

```diff
@@ -250,7 +254,7 @@ def WriteVtable(self):
 
             # Finally, store this vtable in memory for future
             # deduplication:
-            self.vtables[vtKey] = self.Offset()
+            self.vtables.setdefault(vtKeyHash, []).append(self.Offset())
```

Comment:
I'd prefer you to use `if vtKeyHash not in self.vtables:` pattern here, instead of `setdefault`.

Reply:
Why? This is idiomatic Python.

Reply:
@malthe Really? Doesn't `setdefault` evaluate its arguments even if they aren't used? If the argument provided to `setdefault` is heavy to construct, that would create extra garbage.

Reply:
It does create garbage, but since it's refcounted, the cost is negligible. When you write scripts in Python, this is a price you're willing to pay. In return you get readable code.

If Python had generics, it could have a `defaultdict` implementation that wouldn't incur this cost – but it doesn't ...

What you can do is:
```
offsets = self.vtables.get(vtKeyHash, None)
if offsets is None:
    offsets = self.vtables[vtKeyHash] = []
offsets.append(vtKeyHash)
```
But again, the whole point of using Python is convenience and readability, etc.

Reply:
Sorry, saying that "Python is all about readability" is not a good argument for the internal code of a library that tries to be as efficient as possible. The implementation code of these functions can be absolutely horrible looking Python, if it helps speed. The rest of the Python implementation is already very un-typical Python, and so is the public API, frankly.

Reply:
It doesn't help speed in this case unless you're counting every last nanosecond (and if you did, then you wouldn't be using Python). I agree that horrible-looking Python – or even just "less idiomatic" Python – can be justified in hot loops, but this particular block is not part of the would-be hot loop.